crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -584,11 +588,103 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
                 let output = self.object_summary_output(&summary, view_events, view_gas_used);
                 Ok(output)
             }
+            SuiSubcommand::UpgradePackage(UpgradePackageCommand {
+                package,
+                upgrade_capability,
+                gas_budget,
+            }) => {
+                let data = data.ok_or_else(|| {
+                    anyhow::anyhow!(
+                        "Expected a module text block following 'upgrade' starting on lines {}-{}",
+                        start_line,
+                        command_lines_stop
+                    )
+                })?;
+                let (units, warnings_opt) = self.compile_source_units(data.path())?;
```

Comment:
I feel like we should support MVIR too, shouldn't be that much extra? (just a bit more copy pasting)

Reply:
done!